app/src/sharedTest/java/org/oppia/android/app/classroom/ClassroomListFragmentTest.kt

```diff
@@ -0,0 +1,886 @@
+      .assertTextContains("Fractions")
+      .assertTextContains("2 Lessons")
+      .assertIsDisplayed()
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(5)
+      .assertTextContains("Ratios and Proportional Reasoning")
+      .assertTextContains("4 Lessons")
+      .assertIsDisplayed()
+  }
+
+  @Test
+  @Ignore("Temporarily ignored as the test is failing.")
+  fun testFragment_loginProfiles_switchClassrooms_classroomsRetainedIndividually() {
+    profileTestHelper.logIntoAdmin()
+    testCoroutineDispatchers.runCurrent()
+
+    // Click on Science classroom card.
+    composeRule.onNodeWithTag(CLASSROOM_LIST_TEST_TAG).onChildAt(0).performClick()
+    testCoroutineDispatchers.runCurrent()
+
+    // Check that Science classroom's topics are displayed.
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(4)
+      .assertTextContains("First Test Topic")
+      .assertTextContains("3 Lessons")
+      .assertIsDisplayed()
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(5)
+      .assertTextContains("Second Test Topic")
+      .assertTextContains("1 Lesson")
+      .assertIsDisplayed()
+
+    profileTestHelper.logIntoUser()
+    testCoroutineDispatchers.runCurrent()
+
+    // Click on Maths classroom card.
+    composeRule.onNodeWithTag(CLASSROOM_LIST_TEST_TAG).onChildAt(1).performClick()
+    testCoroutineDispatchers.runCurrent()
+
+    // Check that Maths classroom's topics are displayed.
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(4)
+      .assertTextContains("Fractions")
+      .assertTextContains("2 Lessons")
+      .assertIsDisplayed()
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(5)
+      .assertTextContains("Ratios and Proportional Reasoning")
+      .assertTextContains("4 Lessons")
+      .assertIsDisplayed()
+
+    profileTestHelper.logIntoAdmin()
+    testCoroutineDispatchers.runCurrent()
+
+    // Check that Science classroom's topics are displayed.
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(4)
+      .assertTextContains("First Test Topic")
+      .assertTextContains("3 Lessons")
+      .assertIsDisplayed()
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(5)
+      .assertTextContains("Second Test Topic")
+      .assertTextContains("1 Lesson")
+      .assertIsDisplayed()
+
+    profileTestHelper.logIntoUser()
+    testCoroutineDispatchers.runCurrent()
+
+    // Check that Maths classroom's topics are displayed.
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(4)
+      .assertTextContains("Fractions")
+      .assertTextContains("2 Lessons")
+      .assertIsDisplayed()
+    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(5)
+      .assertTextContains("Ratios and Proportional Reasoning")
+      .assertTextContains("4 Lessons")
+      .assertIsDisplayed()
+  }
```

Comment:
```suggestion
  fun testFragment_loginProfiles_switchClassrooms_classroomsRetainedIndividually() {
    profileTestHelper.logIntoAdmin()
    testCoroutineDispatchers.runCurrent()

    // Click on Science classroom card.
    composeRule.onNodeWithTag(CLASSROOM_LIST_TEST_TAG).onChildAt(0).performClick()
    testCoroutineDispatchers.runCurrent()

    // Check that Science classroom's topics are displayed.
    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(4)
      .assertTextContains("First Test Topic")
      .assertTextContains("3 Lessons")
      .assertIsDisplayed()
    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(5)
      .assertTextContains("Second Test Topic")
      .assertTextContains("1 Lesson")
      .assertIsDisplayed()

    // Click on Maths classroom card.
    composeRule.onNodeWithTag(CLASSROOM_LIST_TEST_TAG).onChildAt(1).performClick()
    testCoroutineDispatchers.runCurrent()

    // Check that Maths classroom's topics are displayed.
    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(4)
      .assertTextContains("Fractions")
      .assertTextContains("2 Lessons")
      .assertIsDisplayed()
    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(5)
      .assertTextContains("Ratios and Proportional Reasoning")
      .assertTextContains("4 Lessons")
      .assertIsDisplayed()

    // Click on Science classroom card.
    composeRule.onNodeWithTag(CLASSROOM_LIST_TEST_TAG).onChildAt(0).performClick()
    testCoroutineDispatchers.runCurrent()

    // Check that Science classroom's topics are displayed.
    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(4)
      .assertTextContains("First Test Topic")
      .assertTextContains("3 Lessons")
      .assertIsDisplayed()
    composeRule.onNodeWithTag(CLASSROOM_LIST_SCREEN_TEST_TAG).onChildAt(5)
      .assertTextContains("Second Test Topic")
      .assertTextContains("1 Lesson")
      .assertIsDisplayed()
  }
```
This test is attempting to do an integration test which we can do, but requires a different kind of setup(e.g, simulating different app instances for each user, then a second instance for the second user, then a third instance to verify the first user again).

For the purposes of this PR, I am happy with us verifying that classroom switching works for one user, as per my suggestion. I am unable to invest the time now to setup your intended scenario. your

Reply:
I understand, updated the test per suggestion.